pytorch_lightning/loggers/mlflow.py

```diff
@@ -155,6 +161,14 @@ def experiment(self) -> MlflowClient:
                 )
 
         if self._run_id is None:
+            if self._run_name is not None:
+                self.tags = self.tags or {}
```

Comment:
can we maybe lift this to the init already? So that `self.tags` is always a dict?

Reply:
Thanks. I have a concern about this. The current `self.tags` looks to be a public attribute (it has no prefix '_'), so I worried that the user could set `None` to it after the initialization. 

Note that the document does not say the `tags` are public, so I'm not sure it is intentional: 
https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.loggers.mlflow.html#mlflow-logger

Another minor reason why I put this here is to minimize the side effect. If the `run_name` is not set explicitly, the `self.tags` still be 'None' after this PR. this is the same behavior as the current version.

I could not judge which is better, so any suggestions and comments are welcome!